resources/tests.mjs

```diff
@@ -137,6 +137,60 @@ Suites.push({
     ],
 });
 
+Suites.push({
+    name: "TodoMVC-WebComponents",
+    url: "tentative/todomvc/index.html",
+    tags: ["todomvc"],
+    async prepare(page) {
+        await page.waitForElement("todo-app");
+    },
+    tests: [
+        /* new BenchmarkTestStep(`Adding${numberOfItemsToAdd}Items`, (page) => {
+            const app = page.querySelector("todo-app").callElementMethod("getInstance");
+            const input = app.topbar.todoInput;
+
+            for (let i = 0; i < numberOfItemsToAdd; i++) {
+                input.value = `Something to do ${i}`;
+                input.dispatchEvent(new KeyboardEvent("keyup", { key: "Enter" }));
+            }
+        }),
+        new BenchmarkTestStep("CompletingAllItems", (page) => {
+            const app = page.querySelector("todo-app").callElementMethod("getInstance");
+            const elements = app.list._elements;
+            for (let i = 0; i < numberOfItemsToAdd; i++)
+                elements[i].toggleInput.click();
+        }),
+        new BenchmarkTestStep("DeletingAllItems", (page) => {
+            const app = page.querySelector("todo-app").callElementMethod("getInstance");
+            const elements = app.list._elements;
+            for (let i = 0; i < numberOfItemsToAdd; i++)
+                elements[i].todoButton.click();
+        }), */
+        new BenchmarkTestStep(`Adding${numberOfItemsToAdd}Items`, (page) => {
+            const input = page.querySelector(".new-todo-input", ["todo-app", "todo-topbar"]);
+            for (let i = 0; i < numberOfItemsToAdd; i++) {
+                input.setValue(`Something to do ${i}`);
+                input.dispatchEvent("input");
+                input.enter("keyup");
+            }
+        }),
+        new BenchmarkTestStep("CompletingAllItems", (page) => {
+            const items = page.querySelectorAll("todo-item", ["todo-app", "todo-list"]);
+            for (let i = 0; i < numberOfItemsToAdd; i++){
+                const item = items[i].querySelector(".toggle-todo-input");
```

Comment:
PageElement now also has a querySelector method that I used here. This method looks at the shadow dom if it exists, otherwise a typical selector.